app/controllers/reports_controller.rb

```diff
@@ -31,8 +31,61 @@ def identifiers_to_reconcile
     end
   end
 
+  def pulfa_ark_report
+    authorize! :show, Report
+    if params[:since_date]
+      @resources = ark_report_resources(params[:since_date]) || []
+    end
+
+    respond_to do |format|
+      format.html
+      format.csv do
+        send_data hashes_to_csv(["id", "component_id", "ark", "url"], @resources.map { |r| resource_hash(r) }),
+                  filename: "pulfa-ark-report-#{params[:since_date]}-to-#{Time.zone.today}.csv"
+      end
+    end
+  end
+
   private
 
+    def ark_report_resources(since_date)
+      updated = []
+      query_service.custom_queries.memory_efficient_all(except_models: excluded).each_slice(100) do |records|
+        updated << updated_archival_resources(records, since_date)
+      end
+      updated.flatten!
+    end
+
+    def excluded
+      [Collection, EphemeraBox, EphemeraFolder, EphemeraProject, EphemeraTerm, EphemeraVocabulary, FileSet]
+    end
+
+    def updated_archival_resources(records, since_date)
+      records.select do |r|
+        metadata_id = Array.wrap(r.source_metadata_identifier).first
+        updated = Array.wrap(r.source_metadata_identifier).first
```

Comment:
This isn't right, right? Maybe we need some more tests?

Reply:
Oops — I'll add a test that checks the updated date because it's obviously not being tested now.